src/providers/ldap/sdap_async_autofs.c

```diff
@@ -1371,14 +1372,17 @@ static void sdap_autofs_get_entry_done(struct tevent_req *subreq)
         return;
     }
 
-    ret = sdap_autofs_save_entry(state->id_ctx->be->domain,
-                                 state->opts,
-                                 reply_count != 0 ? reply[0] : NULL,
-                                 state->mapname,
-                                 state->entryname);
-    if (ret != EOK) {
-        tevent_req_error(req, ret);
-        return;
+    /* If other attribute then automountKey is in the distinguished name and
+     * there are multiple entries with different casing then we may get more
+     * then one result. */
+    for (i = 0; i < reply_count; i++) {
+        ret = sdap_autofs_save_entry(state->id_ctx->be->domain, state->opts,
+                                     reply_count != 0 ? reply[i] : NULL,
```

Comment:
I am thinking about this line. Maybe instead of checking if reply_count != 0 for every for() spin we should check it once before loop?

Reply:
I think this is copy-paste :)

Reply:
Thanks, patch is now updated. I added a case when no reply was found so the entry can be removed.